src/pixie/paths.nim

```diff
@@ -1233,43 +1234,122 @@ proc fillShapes(
 
 proc strokeShapes(
   shapes: seq[seq[Vec2]],
-  strokeWidth: float32
+  strokeWidth: float32,
+  lineCap: LineCap,
+  lineJoin: LineJoin,
+  miterAngleLimit = degToRad(28.96)
 ): seq[seq[Vec2]] =
   if strokeWidth == 0:
     return
 
-  let
-    widthLeft = strokeWidth / 2
-    widthRight = strokeWidth / 2
+  let halfStroke = strokeWidth / 2
+
+  proc makeCircle(at: Vec2): seq[Vec2] =
+    var path: Path
+    path.ellipse(at, halfStroke, halfStroke)
+    path.commandsToShapes()[0]
+
+  proc makeRect(at, to: Vec2): seq[Vec2] =
+    # Rectangle corners
+    let
+      tangent = (to - at).normalize()
+      normal = vec2(tangent.y, tangent.x)
+      a = vec2(
+        at.x + normal.x * halfStroke,
+        at.y - normal.y * halfStroke
+      )
+      b = vec2(
+        to.x + normal.x * halfStroke,
+        to.y - normal.y * halfStroke
+      )
+      c = vec2(
+        to.x - normal.x * halfStroke,
+        to.y + normal.y * halfStroke
+      )
+      d = vec2(
+        at.x - normal.x * halfStroke,
+        at.y + normal.y * halfStroke
+      )
+
+    @[a, b, c, d, a]
 
   for shape in shapes:
-    var
-      strokeShape: seq[Vec2]
-      back: seq[Vec2]
-    for segment in shape.segments:
+    var shapeStroke: seq[seq[Vec2]]
+
+    proc makeJoin(prevPos, pos, nextPos: Vec2): seq[Vec2] =
```

Comment:
I think you can move this with the rest of make* shapes. There is no need to have access to shape or shapeStorke.